protocol/swap/types.go

```diff
@@ -0,0 +1,60 @@
+package swap
+
+import (
+	"github.com/athanorlabs/atomic-swap/common/types"
+)
+
+type (
+	Status = types.Status //nolint:revive
+)
+
+// Info contains the details of the swap as well as its status.
+type Info struct {
+	ID             types.Hash // swap offer ID
+	Provides       types.ProvidesCoin
+	ProvidedAmount float64
+	ReceivedAmount float64
+	ExchangeRate   types.ExchangeRate
+	EthAsset       types.EthAsset
+	Status         Status
+	statusCh       <-chan types.Status `json:"-"`
+}
+
+// NewInfo creates a new *Info from the given parameters.
+// Note that the swap ID is the same as the offer ID.
+func NewInfo(
+	id types.Hash,
+	provides types.ProvidesCoin,
+	providedAmount, receivedAmount float64,
+	exchangeRate types.ExchangeRate,
+	ethAsset types.EthAsset,
+	status Status,
+	statusCh <-chan types.Status,
```

Comment:
Do we want this function? I'm not saying we don't, but I'll play devil's advocate here.  If you initialize the `Info` struct directly you get field labeling on all 7 fields. You could change the function `NewInfo` to take `*Info` as a parameter and get labeling, but that would kind of defeat the purpose.

Reply:
I prefer having constructors for types where all fields are required, otherwise it's unclear what fields need to be populated and how imo. also IDEs should show all argument names, so I don't necessarily see the need for labelling